From a7d41cc40357d10cbb84ac463032f4d28af39162 Mon Sep 17 00:00:00 2001
From: Matt Porter <mporter@konsulko.com>
Date: Wed, 13 Dec 2017 12:49:20 -0500
Subject: [PATCH] arm64: dts: renesas: preserve drm HDMI connector naming
 on KF

Kingfisher adds encoders on ports 0 and 2 of the R-Car
display unit. The bare ULCB SK has only an HDMI encoder
in use on port 1. When the system is booted with an SK
dtb, port 1's HDMI encoder is assigned as HDMI-A-1 because
it is the first (and only) encoder present. When booting
a KF dtb, port 0's HDMI encoder preceeds port1 in the dtb due
to incremental ordering of the endpoint port nodes. This causes
the KF HDMI to be assigned HDMI-A-1 and the SK HDMI to be assigned
as HDMI-A-2. In order to preserve the SK's HDMI output naming as
connector HDMI-A-1, reorder the endpoint port nodes so that port1
is first.

Change-Id: Ibbb1975c2383a526a54c257fb7d68d32a042d468
Signed-off-by: Matt Porter <mporter@konsulko.com>
Signed-off-by: Ronan Le Martret <ronan.lemartret@iot.bzh>
---
 arch/arm64/boot/dts/renesas/r8a77951.dtsi | 10 +++++-----
 arch/arm64/boot/dts/renesas/r8a77960.dtsi | 10 +++++-----
 arch/arm64/boot/dts/renesas/r8a77961.dtsi | 10 +++++-----
 3 files changed, 15 insertions(+), 15 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a77951.dtsi b/arch/arm64/boot/dts/renesas/r8a77951.dtsi
index 655a71157a5f..c8aa7be0e33e 100644
--- a/arch/arm64/boot/dts/renesas/r8a77951.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a77951.dtsi
@@ -2373,6 +2373,11 @@ vin7csi41: endpoint@3 {
 						remote-endpoint = <&csi41vin7>;
 					};
 				};
+				port@0 {
+					reg = <0>;
+					du_out_rgb: endpoint {
+					};
+				};
 			};
 		};
 
@@ -3955,11 +3960,6 @@ ports {
 				#address-cells = <1>;
 				#size-cells = <0>;
 
-				port@0 {
-					reg = <0>;
-					du_out_rgb: endpoint {
-					};
-				};
 				port@1 {
 					reg = <1>;
 					du_out_hdmi0: endpoint {
diff --git a/arch/arm64/boot/dts/renesas/r8a77960.dtsi b/arch/arm64/boot/dts/renesas/r8a77960.dtsi
index 1b4099f571bd..22ef3d7047fd 100644
--- a/arch/arm64/boot/dts/renesas/r8a77960.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a77960.dtsi
@@ -2279,6 +2279,11 @@ vin7csi40: endpoint@2 {
 						remote-endpoint = <&csi40vin7>;
 					};
 				};
+				port@0 {
+					reg = <0>;
+					du_out_rgb: endpoint {
+					};
+				};
 			};
 		};
 
@@ -3640,11 +3645,6 @@ ports {
 				#address-cells = <1>;
 				#size-cells = <0>;
 
-				port@0 {
-					reg = <0>;
-					du_out_rgb: endpoint {
-					};
-				};
 				port@1 {
 					reg = <1>;
 					du_out_hdmi0: endpoint {
diff --git a/arch/arm64/boot/dts/renesas/r8a77961.dtsi b/arch/arm64/boot/dts/renesas/r8a77961.dtsi
index ef2cb77f3c32..e1555d3673e2 100644
--- a/arch/arm64/boot/dts/renesas/r8a77961.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a77961.dtsi
@@ -2257,6 +2257,11 @@ vin7csi40: endpoint@2 {
 						remote-endpoint = <&csi40vin7>;
 					};
 				};
+				port@0 {
+					reg = <0>;
+					du_out_rgb: endpoint {
+					};
+				};
 			};
 		};
 
@@ -3434,11 +3439,6 @@ ports {
 				#address-cells = <1>;
 				#size-cells = <0>;
 
-				port@0 {
-					reg = <0>;
-					du_out_rgb: endpoint {
-					};
-				};
 				port@1 {
 					reg = <1>;
 					du_out_hdmi0: endpoint {
-- 
2.31.1

